src/controllers/profile.controller.ts

```diff
@@ -0,0 +1,71 @@
+import { type Request, type Response } from 'express'
+import { dataSource } from '../configs/dbConfig'
+import Profile from '../entities/profile.entity'
+
+export const getProfile = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    res.status(200).json(req.user)
+  } catch (err) {
+    console.error('Error executing query', err)
+    res.status(500).json({ error: err })
+  }
+}
+
+export const updateProfile = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    const user: Profile | undefined = req.user as Profile
+
+    const {
+      primary_email,
```

Comment:
If we are allowing the users to change the primary email, we have to do some additional validation to check whether that email is not associated with another profile

Reply:
primary_email is unique at the DB level. It will throw an error. But that error is not user friendly.